app/src/main/java/com/devshish/internship/presentation/ui/player/PlayerViewModel.kt

```diff
@@ -53,9 +53,10 @@ class PlayerViewModel(
                 _songToPlay.value = song
                 _isPlayerBarVisible.value = true
 
-                // TODO HARDCODED LOGIC
                 viewModelScope.launch {
-                    searchSong.value = repository.searchSongs(song.title).first()
```

Comment:
1. Currently your icon is shown at start - but it should be hidden (I wrote about it in previous pull request, but forgot to check)
2. Move this check into some private method